### ISSUE TITLE
DatePicker: Error in the DatePickerType.Month view!

### ISSUE THREAD

**Issue by true-perfect-code** [ant-design-blazor/ant-design-blazor#3305]
If you pass the date 31 Dec. 2023 to the component via binding and then try to select November in the component DatePicker, this triggers an error and the application crashes.

![image](https://github.com/ant-design-blazor/ant-design-blazor/assets/45635963/e4906edd-83b7-443a-894e-440ac4d2e178)

```
System.ArgumentOutOfRangeException: Year, Month, and Day parameters describe an un-representable DateTime.
   at System.DateTime..ctor(Int32 year, Int32 month, Int32 day, Int32 hour, Int32 minute, Int32 second)
   at AntDesign.DateHelper.CombineNewDate(DateTime date, Nullable`1 year, Nullable`1 month, Nullable`1 day, Nullable`1 hour, Nullable`1 minute, Nullable`1 second)
   at AntDesign.DatePickerPanelBase`1.CombineNewShowDate(Nullable`1 year, Nullable`1 month, Nullable`1 day, Nullable`1 hour, Nullable`1 minute, Nullable`1 second)
   at AntDesign.DatePickerPanelBase`1.OnSelectMonth(DateTime date)
   at AntDesign.Internal.DatePickerMonthPanel`1.<BuildRenderTree>b__0_5(DateTime date)
   at AntDesign.Internal.DatePickerTemplate`1.<>c__DisplayClass87_2.<BuildRenderTree>b__4(MouseEventArgs e)
   at InvokeStub_Action`1.Invoke(Object, Object, IntPtr*)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
--- End of stack trace from previous location ---
```

**The reason for this is probably that you set the day within the DatePicker component on the basis of the binding date, so if you come up with 31, then it tries to integrate 31 into the date for November, which of course does not work.**

This problem of setting the day can be solved relatively easily programmatically in the datePicker component by programming it in such a way that you can determine another parameter with which you can decide whether the day of the binding date is to be taken over (is now also programmed in this way), but that you offer 3 more options, i.e.:
- sets the first day, i.e. 1
- set the last day (28, 29, 30 or 31)
- you set the current day (e.g. if today is 15 June, then day 15 comes). Here, however, you still have to catch the possible error and reduce the day if the range is outside (e.g. November has 30 days).

Regards
perfect-code

**Comment by ElderJames** [comment]
Hello @true-perfect-code , I have tried on the current version, and it works fine.

![datepicker-month](https://github.com/ant-design-blazor/ant-design-blazor/assets/7550366/b3610dfa-07fd-44ec-bcdc-aa5ac6ebd355)

**Comment by Alexbits** [comment]
@ElderJames @true-perfect-code  This was fixed by #3193

**Comment by true-perfect-code** [comment]
Thanks @Alexbits 

However, the suggestion of an additional parameter could be interesting for a later enhancement.
As it is programmed now, the last day in a month is always taken.
